mutiply_by_2 with for...in loop

**V version:** V 0.1.25 050ec5d
**OS:** Linux alexp-pc 5.3.0-28-generic #30~18.04.1-Ubuntu SMP Fri Jan 17 06:14:09 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

**What did you do?**
```
fn double_up(a mut []int) {
	for i := 0; i < a.len; i++ {
		a[i] = a[i]*2
	}
}

fn double_up_v2(a mut []int) {
	for i, val in a {
		a[i] = a[i]*2 // or val*2, doesn't matter
	}
}

fn main() {
	mut arr := [1,2,3,4,5,6,7,8,9,10]
	double_up(mut arr)
	double_up_v2(mut arr)
	print(arr)
}
```
**What did you expect to see?**
expected to see result like `arr.map(it*4)`
 
**What did you see instead?**
```
/home/alexp/.cache/v/main.tmp.c: In function ‘main__double_up_v2’:
/home/alexp/.cache/v/main.tmp.c:6816:26: error: ‘tmp3’ is a pointer; did you mean to use ‘->’?
  for (int i = 0; i < tmp3.len; i++) {
                          ^
                          ->
/home/alexp/.cache/v/main.tmp.c:6817:28: error: ‘tmp3’ is a pointer; did you mean to use ‘->’?
  int* val = ((int* *) tmp3 . data)[i];
                            ^
                            ->
/home/alexp/.cache/v/main.tmp.c:6819:38: error: invalid operands to binary * (have ‘int *’ and ‘int’)
 array_set( a , i , & (int []) {  val * 2 }) ;
                                      ^
...
==================
(Use `v -cg` to print the entire error message)

V error: C error.
```

--- a/vlib/v/gen/cgen.v
+++ b/vlib/v/gen/cgen.v
@@ -481,17 +481,19 @@ fn (g mut Gen) for_in(it ast.ForInStmt) {
 		g.expr(it.cond)
 		cond_type_is_ptr := table.type_is_ptr(it.cond_type)
 		if cond_type_is_ptr {
-			g.writeln('->len; $i++) {')
+			g.writeln('->')
 		} else {
-			g.writeln('.len; $i++) {')
+			g.writeln('.')
 		}
+		g.write('len; $i++) {')
 		g.write('\t$styp $it.val_var = (($styp*)')
 		g.expr(it.cond)
 		if cond_type_is_ptr {
-			g.writeln('->data)[$i];')
+			g.writeln('->')
 		} else {
-			g.writeln('.data)[$i];')
+			g.writeln('.')
 		}
+		g.write('data)[$i];')
 		g.stmts(it.stmts)
 		g.writeln('}')
 	}

--- a/vlib/v/gen/cgen.v
+++ b/vlib/v/gen/cgen.v
@@ -479,10 +479,19 @@ fn (g mut Gen) for_in(it ast.ForInStmt) {
 		styp := g.typ(it.val_type)
 		g.write('for (int $i = 0; $i < ')
 		g.expr(it.cond)
-		g.writeln('.len; $i++) {')
+		cond_type_is_ptr := table.type_is_ptr(it.cond_type)
+		if cond_type_is_ptr {
+			g.writeln('->len; $i++) {')
+		} else {
+			g.writeln('.len; $i++) {')
+		}
 		g.write('\t$styp $it.val_var = (($styp*)')
 		g.expr(it.cond)
-		g.writeln('.data)[$i];')
+		if cond_type_is_ptr {
+			g.writeln('->data)[$i];')
+		} else {
+			g.writeln('.data)[$i];')
+		}
 		g.stmts(it.stmts)
 		g.writeln('}')
 	}
@@ -1705,6 +1714,9 @@ fn (g mut Gen) index_expr(node ast.IndexExpr) {
 				// `x[0] *= y`
 				if g.assign_op in [.mult_assign] {
 					g.write('*($elem_type_str*)array_get(')
+					if left_is_ptr {
+						g.write('*')
+					}
 					g.expr(node.left)
 					g.write(', ')
 					g.expr(node.index)
@@ -1721,6 +1733,9 @@ fn (g mut Gen) index_expr(node ast.IndexExpr) {
 			}
 			else {
 				g.write('(*($elem_type_str*)array_get(')
+				if left_is_ptr {
+					g.write('*')
+				}
 				g.expr(node.left)
 				g.write(', ')
 				g.expr(node.index)

